frontend/packages/console-shared/src/utils/pod-utils.ts

```diff
@@ -0,0 +1,311 @@
+  Idle: '#FFFFFF',
+  'Autoscaled to 0': '#FFFFFF',
+};
+
+export const podStatus = Object.keys(podColor);
+
+function isContainerFailedFilter(containerStatus) {
+  return containerStatus.state.terminated && containerStatus.state.terminated.exitCode !== 0;
+}
+
+function isContainerLoopingFilter(containerStatus) {
+  return (
+    containerStatus.state.waiting && containerStatus.state.waiting.reason === 'CrashLoopBackOff'
+  );
+}
+
+function numContainersReadyFilter(pod) {
+  let numReady = 0;
+  _.forEach(pod.status.containerStatuses, function(status) {
+    if (status.ready) {
+      numReady++;
+    }
+  });
+  return numReady;
+}
+
+function isReady(pod) {
+  const numReady = numContainersReadyFilter(pod);
+  const total = _.size(pod.spec.containers);
+
+  return numReady === total;
+}
+
+function podWarnings(pod) {
+  if (pod.status.phase === 'Running' && pod.status.containerStatuses) {
+    return _.map(pod.status.containerStatuses, function(containerStatus) {
+      if (!containerStatus.state) {
+        return null;
+      }
+
+      if (isContainerFailedFilter(containerStatus)) {
+        if (_.has(pod, ['metadata', 'deletionTimestamp'])) {
+          return 'Failed';
+        }
+        return 'Warning';
+      }
+      if (isContainerLoopingFilter(containerStatus)) {
+        return 'Failed';
+      }
+      return null;
+    }).filter((x) => x);
+  }
+  return null;
+}
+
+export function getPodStatus(pod) {
+  if (_.has(pod, ['metadata', 'deletionTimestamp'])) {
+    return 'Terminating';
+  }
+  const warnings = podWarnings(pod);
+  if (warnings !== null && warnings.length) {
+    if (warnings.includes('Failed')) {
+      return 'Failed';
+    }
+    return 'Warning';
+  }
+  if (pod.status.phase === 'Running' && !isReady(pod)) {
+    return 'Not Ready';
+  }
+  return _.get(pod, 'status.phase', 'Unknown');
+}
+
+export const calculateRadius = (size: number) => {
+  const radius = size / 2;
+  const podStatusStrokeWidth = (8 / 104) * size;
+  const podStatusInset = (5 / 104) * size;
+  const podStatusOuterRadius = radius - podStatusInset;
+  const podStatusInnerRadius = podStatusOuterRadius - podStatusStrokeWidth;
+  const decoratorRadius = radius * 0.25;
+
+  return {
+    radius,
+    podStatusInnerRadius,
+    podStatusOuterRadius,
+    decoratorRadius,
+  };
+};
+
+export const checkPodEditAccess = (
+  resource: K8sResourceKind,
+  resourceKind: K8sKind,
+  impersonate: string,
+): Promise<SelfSubjectAccessReviewKind> => {
+  if (_.isEmpty(resource) || !resourceKind) {
+    return Promise.resolve(null);
+  }
+  const { name, namespace } = resource.metadata;
+  const resourceAttributes: AccessReviewResourceAttributes = {
+    group: resourceKind.apiGroup,
+    resource: resourceKind.plural,
+    verb: 'patch',
+    name,
+    namespace,
+  };
+  return checkAccess(resourceAttributes, impersonate);
+};
+
+export class TransformPodData {
+  constructor(private resources: PodDataResources) {
+    this.resources = resources;
```

Comment:
The line above is technically redundant, TypeScript ["parameter properties"](https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties) consolidate declaration and assignment into one location - constructor parameter.

So the code above is equivalent to:

```ts
constructor(private resources: PodDataResources) {}
```

If ESLint warns about an empty constructor body, we need to configure it to understand above TypeScript feature, or suppress the warning with e.g. `// eslint-disable-next-line`.

Reply:
Yes, it was giving ```ESLint``` error that is I added this line in the constructor.

Reply:
In that case, I'd suppress that ESLint error instead of having redundant code to make ESLint happy.